TransformationSystem/Agent/MCExtensionAgent.py

```diff
@@ -1,108 +1,117 @@
-    res = self.transClient.getTransformationTaskStats( transID )
+    res = self.transClient.getTransformationTaskStats(transID)
     if not res['OK']:
       if res['Message'] != 'No records found':
-        gLogger.error( "Failed to get task statistics", "%s %s" % ( transID, res['Message'] ) )
+        gLogger.error("Failed to get task statistics", "%s %s" % (transID, res['Message']))
         return res
       else:
         statusDict = {}
     else:
       statusDict = res['Value']
-    gLogger.verbose( "Current task count for transformation %d" % transID )
-    for status in sorted( statusDict.keys() ):
+    gLogger.verbose("Current task count for transformation %d" % transID)
+    for status in sorted(statusDict.keys()):
       statusCount = statusDict[status]
-      gLogger.verbose( "%s : %s" % ( status.ljust( 20 ), str( statusCount ).rjust( 8 ) ) )
+      gLogger.verbose("%s : %s" % (status.ljust(20), str(statusCount).rjust(8)))
     # Determine the number of tasks to be created
-    numberOfTasks = self._calculateTaskNumber( maxTasks, statusDict )
+    numberOfTasks = self._calculateTaskNumber(maxTasks, statusDict)
     if not numberOfTasks:
-      gLogger.info( "No tasks required for transformation %d" % transID )
+      gLogger.info("No tasks required for transformation %d" % transID)
       return S_OK()
     # Extend the transformation by the determined number of tasks
-    res = self.transClient.extendTransformation( transID, numberOfTasks )
+    res = self.transClient.extendTransformation(transID, numberOfTasks)
     if not res['OK']:
-      gLogger.error( "Failed to extend transformation", "%s %s" % ( transID, res['Message'] ) )
+      gLogger.error("Failed to extend transformation", "%s %s" % (transID, res['Message']))
       return res
-    gLogger.info( "Successfully extended transformation %d by %d tasks" % ( transID, numberOfTasks ) )
+    gLogger.info("Successfully extended transformation %d by %d tasks" % (transID, numberOfTasks))
     return S_OK()
 
-  def _calculateTaskNumber( self, maxTasks, statusDict ):
+  def _calculateTaskNumber(self, maxTasks, statusDict):
     ''' Utility function
     '''
-    done = statusDict.get( 'Done', 0 )
-    failed = statusDict.get( 'Failed', 0 )
-    waiting = statusDict.get( 'Waiting', 0 )
+    done = statusDict.get('Done', 0)
+    failed = statusDict.get('Failed', 0)
+    waiting = statusDict.get('Waiting', 0)
     total = statusDict.get('TotalCreated', 0)
     # If the failure rate is higher than acceptable
-    if ( total != 0 ) and ( ( 100.0 * float( failed ) / float( total ) ) > self.maxFailRate ):
+    if (total != 0) and ((100.0 * float(failed) / float(total)) > self.maxFailRate):
       return 0
     # If we already have enough completed jobs
     if done >= maxTasks:
       return 0
     if waiting > self.maxWaitingJobs:
       return 0
-    numberOfTasks = maxTasks - ( total - failed )
+    numberOfTasks = maxTasks - (total - failed)
```

Comment:
Can you add
```python
if numberOfTasks < 0:
  return 0
```
That happens when people extend the transformation manually instead of increasing MaxNumberOfTasks

Reply:
Done.